src/uipath/tracing/_utils.py

```diff
@@ -1,321 +1,331 @@
-            # Create a parameter mapping by partially binding the arguments
-
-            parameter_binding = signature.bind_partial(*args, **kwargs)
-
-            # Fill in default values for any unspecified parameters
-            parameter_binding.apply_defaults()
-
-            # Extract the input parameters, skipping special Python parameters
-            result = {}
-            for name, value in parameter_binding.arguments.items():
-                # Skip class and instance references
-                if name in ("self", "cls"):
-                    continue
-
-                # Handle **kwargs parameters specially
-                param_info = signature.parameters.get(name)
-                if param_info and param_info.kind == inspect.Parameter.VAR_KEYWORD:
-                    # Flatten nested kwargs directly into the result
-                    if isinstance(value, dict):
-                        result.update(value)
-                else:
-                    # Regular parameter
-                    result[name] = value
-
-            return result
-        except Exception as e:
-            logger.warning(
-                f"Error formatting arguments for trace: {e}. Using args and kwargs directly."
-            )
-            return {"args": args, "kwargs": kwargs}
+import inspect
+import json
+import logging
+import os
+import random
+import uuid
+from dataclasses import asdict, dataclass, field, is_dataclass
+from datetime import datetime, timezone
+from enum import Enum
+from os import environ as env
+from typing import Any, Dict, Optional
+from zoneinfo import ZoneInfo
+
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.trace import StatusCode
+from pydantic import BaseModel
+
+logger = logging.getLogger(__name__)
+
+
+def _simple_serialize_defaults(obj):
+    # Handle Pydantic BaseModel instances
+    if hasattr(obj, "model_dump") and not isinstance(obj, type):
+        return obj.model_dump(exclude_none=True, mode="json")
+
+    # Handle classes - convert to schema representation
+    if isinstance(obj, type) and issubclass(obj, BaseModel):
+        return {
+            "__class__": obj.__name__,
+            "__module__": obj.__module__,
+            "schema": obj.model_json_schema(),
+        }
+    if hasattr(obj, "dict") and not isinstance(obj, type):
+        return obj.dict()
+    if hasattr(obj, "to_dict") and not isinstance(obj, type):
+        return obj.to_dict()
+
+    # Handle dataclasses
+    if is_dataclass(obj) and not isinstance(obj, type):
+        return asdict(obj)
+
+    # Handle enums
+    if isinstance(obj, Enum):
+        return obj.value
+
+    if isinstance(obj, (set, tuple)):
+        if hasattr(obj, "_asdict") and callable(obj._asdict):
+            return obj._asdict()
+        return list(obj)
+
+    if isinstance(obj, datetime):
+        return obj.isoformat()
+
+    if isinstance(obj, (timezone, ZoneInfo)):
+        return obj.tzname(None)
+
+    return str(obj)
```

Comment:
Shouldn't we allow primitives `None`, `int`, `float`, `bool` to pass through without it being considered a string?